Main test fails with mismatching `project_root_count`

When running `main_tests.go` locally against [v1.131.0](https://github.com/wakatime/wakatime-cli/releases/tag/v1.131.0), numerous test cases fail, with a mismatching `project_root_count`. We are expecting 5, but for some reason, `project_root_count` is set to 6. 

I am packaging the latest release to Arch Linux, and this happens when we execute the tests:

```
build() {
  # provide date but make it reproducible
  local _date="$(date --utc --date="@${SOURCE_DATE_EPOCH:-$(date +%s)}" '+%Y-%m-%dT%H:%M:%S %Z')"
  # we don't care about the commit as we are building from source tarball
  local _commit="0000000"
  local _repo="${url#https://}"

  cd "$_name-$pkgver"

  export CGO_CPPFLAGS="${CPPFLAGS}"
  export CGO_CFLAGS="${CFLAGS}"
  export CGO_CXXFLAGS="${CXXFLAGS}"
  export CGO_LDFLAGS="${LDFLAGS}"

  go build -o build/$pkgname \
    -trimpath \
    -buildmode=pie \
    -mod=readonly \
    -modcacherw \
    -ldflags "-linkmode external -extldflags '${LDFLAGS}' -X '${_repo}/pkg/version.OS=linux' -X '${_repo}/pkg/version.Arch=amd64' -X '${_repo}/pkg/version.BuildDate=${_date}' -X '${_repo}/pkg/version.Commit=${_commit}' -X '${_repo}/pkg/version.Version=${pkgver}'"
}

check () {
  cd "$_name-$pkgver"
  go test -tags=integration ./main_test.go
}
```

**Environment**:

- OS: Linux (Archlinux)
- Platform: amd64

**Logs**:

```
...
--- FAIL: TestPrintOfflineHeartbeats (0.12s)
    main_test.go:1644: 
        	Error Trace:	/build/wakatime/src/wakatime-cli-1.131.0/main_test.go:1644
        	Error:      	Not equal: 
        	            	expected: "[{\"category\":\"writing tests\",\"cursorpos\":12,\"dependencies\":[\"os\"],\"entity\":\"/build/wakatime/src/wakatime-cli-1.131.0/testdata/main.go\",\"type\":\"file\",\"is_write\":true,\"language\":\"Go\",\"lineno\":42,\"lines\":100,\"project\":\"wakatime-cli\",\"project_root_count\":5,\"time\":1585598059,\"user_agent\":\"wakatime/1.131.0 (linux-6.16.3-arch1-1-unknown) go1.24.6 Unknown/0\"}]"
        	            	actual  : "[{\"category\":\"writing tests\",\"cursorpos\":12,\"dependencies\":[\"os\"],\"entity\":\"/build/wakatime/src/wakatime-cli-1.131.0/testdata/main.go\",\"type\":\"file\",\"is_write\":true,\"language\":\"Go\",\"lineno\":42,\"lines\":100,\"project\":\"wakatime-cli\",\"project_root_count\":6,\"time\":1585598059,\"user_agent\":\"wakatime/1.131.0 (linux-6.16.3-arch1-1-unknown) go1.24.6 Unknown/0\"}]"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-[{"category":"writing tests","cursorpos":12,"dependencies":["os"],"entity":"/build/wakatime/src/wakatime-cli-1.131.0/testdata/main.go","type":"file","is_write":true,"language":"Go","lineno":42,"lines":100,"project":"wakatime-cli","project_root_count":5,"time":1585598059,"user_agent":"wakatime/1.131.0 (linux-6.16.3-arch1-1-unknown) go1.24.6 Unknown/0"}]
        	            	+[{"category":"writing tests","cursorpos":12,"dependencies":["os"],"entity":"/build/wakatime/src/wakatime-cli-1.131.0/testdata/main.go","type":"file","is_write":true,"language":"Go","lineno":42,"lines":100,"project":"wakatime-cli","project_root_count":6,"time":1585598059,"user_agent":"wakatime/1.131.0 (linux-6.16.3-arch1-1-unknown) go1.24.6 Unknown/0"}]
        	Test:       	TestPrintOfflineHeartbeats
...
```

More failed logs at: https://gist.github.com/benmezger/ca93da3f79eddb696b76b3d3dad7e6f7

--- a/cmd/fileexperts/fileexperts_test.go
+++ b/cmd/fileexperts/fileexperts_test.go
@@ -32,7 +32,7 @@ func TestFileExperts(t *testing.T) {
 		numCalls int
 	)
 
-	projectFolder, err := filepath.Abs("../..")
+	projectFolder, err := filepath.Abs("testdata")
 	require.NoError(t, err)
 
 	subfolders := project.CountSlashesInProjectFolder(projectFolder)
@@ -85,7 +85,7 @@ func TestFileExperts(t *testing.T) {
 	v.Set("key", "00000000-0000-4000-8000-000000000000")
 	v.Set("api-url", testServerURL)
 	v.Set("plugin", plugin)
-	v.Set("project", "wakatime-cli")
+	v.Set("projectmap..*", "wakatime-cli")
 	v.Set("entity", "testdata/main.go")
 
 	output, err := fileexperts.FileExperts(t.Context(), v)
@@ -143,6 +143,7 @@ func TestFileExperts_ErrApi(t *testing.T) {
 	v.Set("key", "00000000-0000-4000-8000-000000000000")
 	v.Set("api-url", testServerURL)
 	v.Set("entity", "testdata/main.go")
+	v.Set("projectmap..*", "wakatime-cli")
 
 	_, err := fileexperts.FileExperts(t.Context(), v)
 	require.Error(t, err)
@@ -177,6 +178,7 @@ func TestFileExperts_ErrAuth(t *testing.T) {
 	v.Set("key", "00000000-0000-4000-8000-000000000000")
 	v.Set("api-url", testServerURL)
 	v.Set("entity", "testdata/main.go")
+	v.Set("projectmap..*", "wakatime-cli")
 
 	_, err := fileexperts.FileExperts(t.Context(), v)
 	require.Error(t, err)
@@ -210,6 +212,7 @@ func TestFileExperts_ErrBadRequest(t *testing.T) {
 	v.Set("key", "00000000-0000-4000-8000-000000000000")
 	v.Set("api-url", testServerURL)
 	v.Set("entity", "testdata/main.go")
+	v.Set("projectmap..*", "wakatime-cli")
 
 	_, err := fileexperts.FileExperts(t.Context(), v)
 	require.Error(t, err)

--- a/cmd/heartbeat/heartbeat_test.go
+++ b/cmd/heartbeat/heartbeat_test.go
@@ -46,7 +46,7 @@ func TestSendHeartbeats(t *testing.T) {
 		numCalls int
 	)
 
-	projectFolder, err := filepath.Abs("../..")
+	projectFolder, err := filepath.Abs("testdata")
 	require.NoError(t, err)
 
 	tmpFile, err := os.CreateTemp(t.TempDir(), "wakatime-config")
@@ -117,7 +117,7 @@ func TestSendHeartbeats(t *testing.T) {
 	v.Set("language", "Go")
 	v.Set("alternate-language", "Golang")
 	v.Set("hide-branch-names", true)
-	v.Set("project", "wakatime-cli")
+	v.Set("projectmap..*", "wakatime-cli")
 	v.Set("lineno", 13)
 	v.Set("local-file", "testdata/localfile.go")
 	v.Set("plugin", plugin)
@@ -178,7 +178,7 @@ func TestSendHeartbeats_RateLimited(t *testing.T) {
 	v.Set("language", "Go")
 	v.Set("alternate-language", "Golang")
 	v.Set("hide-branch-names", true)
-	v.Set("project", "wakatime-cli")
+	v.Set("projectmap..*", "wakatime-cli")
 	v.Set("lineno", 13)
 	v.Set("local-file", "testdata/localfile.go")
 	v.Set("plugin", plugin)
@@ -286,7 +286,7 @@ func TestSendHeartbeats_ExtraHeartbeats(t *testing.T) {
 
 	ctx := t.Context()
 
-	projectFolder, err := filepath.Abs("../..")
+	projectFolder, err := filepath.Abs("testdata")
 	require.NoError(t, err)
 
 	subfolders := project.CountSlashesInProjectFolder(projectFolder)
@@ -395,7 +395,7 @@ func TestSendHeartbeats_ExtraHeartbeats(t *testing.T) {
 	v.Set("extra-heartbeats", true)
 	v.Set("key", "00000000-0000-4000-8000-000000000000")
 	v.Set("hide-branch-names", true)
-	v.Set("project", "wakatime-cli")
+	v.Set("projectmap..*", "wakatime-cli")
 	v.Set("language", "Go")
 	v.Set("alternate-language", "Golang")
 	v.Set("lineno", 2)
@@ -433,7 +433,7 @@ func TestSendHeartbeats_ExtraHeartbeatsNestedError(t *testing.T) {
 
 	ctx := t.Context()
 
-	projectFolder, err := filepath.Abs("../..")
+	projectFolder, err := filepath.Abs("testdata")
 	require.NoError(t, err)
 
 	subfolders := project.CountSlashesInProjectFolder(projectFolder)
@@ -549,7 +549,7 @@ func TestSendHeartbeats_ExtraHeartbeatsNestedError(t *testing.T) {
 	v.Set("extra-heartbeats", true)
 	v.Set("key", "00000000-0000-4000-8000-000000000000")
 	v.Set("hide-branch-names", true)
-	v.Set("project", "wakatime-cli")
+	v.Set("projectmap..*", "wakatime-cli")
 	v.Set("language", "Go")
 	v.Set("alternate-language", "Golang")
 	v.Set("lineno", 2)
@@ -651,7 +651,7 @@ func TestSendHeartbeats_ExtraHeartbeats_Sanitize(t *testing.T) {
 	v.Set("key", "00000000-0000-4000-8000-000000000000")
 	v.Set("hide-branch-names", true)
 	v.Set("hide-file-names", true)
-	v.Set("project", "wakatime-cli")
+	v.Set("projectmap..*", "wakatime-cli")
 	v.Set("language", "Go")
 	v.Set("alternate-language", "Golang")
 	v.Set("lineno", 13)
@@ -782,7 +782,7 @@ func TestSendHeartbeats_ExtraHeartbeatsIsUnsavedEntity(t *testing.T) {
 
 	ctx := t.Context()
 
-	projectFolder, err := filepath.Abs("../..")
+	projectFolder, err := filepath.Abs(".")
 	require.NoError(t, err)
 
 	subfolders := project.CountSlashesInProjectFolder(projectFolder)
@@ -812,7 +812,7 @@ func TestSendHeartbeats_ExtraHeartbeatsIsUnsavedEntity(t *testing.T) {
 			string(expectedBody),
 			entities[0].Entity, subfolders, userAgent,
 			entities[1].Entity, subfolders, userAgent,
-			entities[2].Entity, subfolders, userAgent,
+			entities[2].Entity, subfolders+1, userAgent,
 		)
 
 		assert.JSONEq(t, expectedBodyStr, string(body))
@@ -873,7 +873,7 @@ func TestSendHeartbeats_ExtraHeartbeatsIsUnsavedEntity(t *testing.T) {
 	v.Set("key", "00000000-0000-4000-8000-000000000000")
 	v.Set("language", "Go")
 	v.Set("alternate-language", "Golang")
-	v.Set("project", "wakatime-cli")
+	v.Set("projectmap..*", "wakatime-cli")
 	v.Set("hide-branch-names", true)
 	v.Set("lineno", 11)
 	v.Set("lines-in-file", 91)
@@ -917,7 +917,7 @@ func TestSendHeartbeats_NonExistingExtraHeartbeatsEntity(t *testing.T) {
 
 	ctx := t.Context()
 
-	projectFolder, err := filepath.Abs("../..")
+	projectFolder, err := filepath.Abs("testdata")
 	require.NoError(t, err)
 
 	subfolders := project.CountSlashesInProjectFolder(projectFolder)
@@ -1001,7 +1001,7 @@ func TestSendHeartbeats_NonExistingExtraHeartbeatsEntity(t *testing.T) {
 	v.Set("entity", "testdata/main.go")
 	v.Set("entity-type", "file")
 	v.Set("hide-branch-names", true)
-	v.Set("project", "wakatime-cli")
+	v.Set("projectmap..*", "wakatime-cli")
 	v.Set("extra-heartbeats", true)
 	v.Set("key", "00000000-0000-4000-8000-000000000000")
 	v.Set("plugin", plugin)

--- a/pkg/project/project.go
+++ b/pkg/project/project.go
@@ -648,11 +648,11 @@ func CountSlashesInProjectFolder(directory string) int {
 	directory = windows.FormatFilePath(directory)
 
 	// Add trailing slash if not present.
-	if !strings.HasSuffix(directory, `/`) {
-		directory += `/`
+	if !strings.HasSuffix(directory, "/") {
+		directory += "/"
 	}
 
-	return strings.Count(directory, `/`)
+	return strings.Count(directory, "/")
 }
 
 // FindFileOrDirectory searches current and all parent folders for a file or directory named `filename`.

